src/main/java/com/codeborne/selenide/SelenideElement.java

```diff
@@ -39,6 +39,29 @@
    */
   SelenideElement setValue(String text);
 
+  /**
+   *
+   * <b>Implementation details:</b>
+   *
+   * <p>If Configuration.versatileSetValue is true, can work as 'selectOptionByValue', 'selectRadio'</p>
+   *
+   * <p>If Configuration.fastSetValue is true, sets value by javascript instead of using Selenium built-in "sendKey" function
+   * and trigger "focus", "keydown", "keypress", "input", "keyup", "change" events.
+   *
+   * <p>In other case behavior will be:
+   * <pre>
+   * 1. WebElement.clear()
+   * 2. WebElement.sendKeys(text)
+   * 3. Trigger change event
+   * </pre>
+   *
+   * @param hiddenString Useful for entering passwords or sensitive data. HiddenString plainTextValue to enter into the text field or set by value for select/radio,
+   * HiddenString toStringValue to log and report.
+   *
+   * @see com.codeborne.selenide.commands.SetValue
+   */
+  SelenideElement setValue(HiddenString hiddenString);
```

Comment:
By the way in test you will pass string in open format such as `new HiddenString("asdf", "***")` and looks like there are no reason to do it in such way.
Again - add tests and proper documentation

Reply:
You can get passwords from password manager (e.g. HashiCorp Vault) and not store them in git.

This is our use case and that is why we want to remove plaintext passwords also from logs/reports,
so that anybody can see logs/reports (e.g. Allure report) without leaking sensitive information.

Reply:
Please see Vault use case and example in Initial comment > Proposed changes.